conn.go

```diff
@@ -415,15 +421,28 @@ func (c *Conn) Close() error {
 	<-c.txDone
 	<-c.rxDone
 
-	var connErr *ConnError
-	if errors.As(c.doneErr, &connErr) && connErr.RemoteErr == nil && connErr.inner == nil {
-		// an empty ConnectionError means the connection was closed by the caller
+	return c.closedErr()
+}
+
+// Done returns a channel that's closed when Conn is closed.
+func (c *Conn) Done() <-chan struct{} {
+	return c.done
+}
+
+// If Done is not yet closed, Err returns nil.
+// If Done is closed, Err returns nil or an error explaining why.
+// A nil error indicates that [Close] was called and there
```

Comment:
I originally wrote it like this so that `Err()` has the same semantics as `Close()`. But now I'm wondering if that's desirable. A return value of `nil` is somewhat ambiguous, i.e. either `Conn` hasn't been closed or it was cleanly closed. Of course, one shouldn't be looking at the return value from `Err()` unless the channel returned by `Done()` has been closed.

Reply:
I think your way is better - 'nil', and it matches what context does.

Reply:
I added an example detailing how these APIs should be used and what the return value from `Err()` can mean.